File: Domain/Todo.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Todo
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Comment { get; set; }
        public Category Category { get; set; }
        public int Priority { get; set; }
        public DateTime Created_At { get; set; }
        public DateTime Updated_At { get; set; }
        public DateTime Finish_Time { get; set; }
        public bool Done { get; set; }

        public ICollection<TodoParticipant> Participants { get; set; } = new List<TodoParticipant>();

    }
}

File: todolist-client/src/app/models/todo.ts
import { Profile } from "./profile";
import { Priority } from "./priority";

export interface Todo {
  id: string;
  title: string;
  description: string;
  comment: string;
  priority?: any;
  created_At: Date | null;
  updated_At: Date | null;
  finish_Time: Date | null;
  done: boolean;
  ownerName?: string;
  participants?: Profile[];
  owner?: Profile;
  isOwner?: boolean;
  category?: any;
}

export interface TodoWithParticipants extends Todo {
  ownerName?: string;
  participants?: Profile[];
  owner?: Profile;
  category?: any;
  priority?: any;
}

export class TodoDTO implements Todo {
  id: string;
  title: string;
  description: string;
  comment: string;
  priority: Priority;
  created_At: Date | null;
  updated_At: Date | null;
  finish_Time: Date | null;
  done: boolean;

  constructor(todo: Todo) {
    this.id = todo.id;
    this.title = todo.title;
    this.description = todo.description;
    this.comment = todo.comment;
    this.priority = todo.priority;
    this.created_At = todo.created_At;
    this.updated_At = todo.updated_At;
    this.finish_Time = todo.finish_Time;
    this.done = todo.done;
  }
}
